server.py

```diff
@@ -1,58 +1,78 @@
-import threading
-from flask import Flask
-from flask_socketio import SocketIO, emit
-from canvas_board import CanvasBoard, CanvasBoardEncoder
+import random, string
+from flask import Flask, jsonify
+from flask_socketio import SocketIO, emit, join_room, leave_room
+from canvas_board import CanvasBoard
+from json import dumps
 
 # constants
 PORT = 8080
 HEIGHT = 500
 WIDTH = 500
+ID_LENGTH = 7
 
 # create Flask object
-app = Flask(__name__)
+app = Flask(__name__, template_folder='static/', static_folder='static/')
 app.config['SECRET_KEY'] = 'secret!'
 socketio = SocketIO(app, cors_allowed_origins='*')
 
 # server class
 class Server:
-    def __init__(self, imagedata):
-        self.board = CanvasBoard(imagedata)
-        self.lock = threading.Lock()
-
-    def update_board(self, diffs: list):
-        self.lock.acquire()
-        self.board.update_board(diffs)
-        self.lock.release()
+    def __init__(self, boards: 'dict that maps room id to CanvasBoard'):
+        self.boards = boards
 
+    def update_board(self, diffs: list, room_id: str):
+        self.boards[room_id].update_board(diffs)
 
 # instantiate server class for board state
 imagedata = {
     'width': WIDTH,
     'height': HEIGHT,
     'data': [0 for i in range(4 * WIDTH * HEIGHT)]
 }

```

Comment:
Since this was used to instantiate a new board, we should move this into CanvasBoard class (along with the appropriate constants).

Better yet, you can keep the constants and instead create a new static method in CanvasBoard to create a new board.  Something like:

```
def __init__(self, width, height):
    self.width = width
    self.height = height
    self.data = # initialize empty board
    self.lock = Thread.lock()

def create_board(width, height):
    return CanvasBoard(width, height)
```
This way we can control how a board is created.